Assets/Scripts/ECS/Events/DeathEventSystem.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using UnityEngine;
+using Unity.Jobs;
+using Unity.Entities;
+using Ecosystem.ECS.Hybrid;
+using Unity.Mathematics;
+using Unity.Transforms;
+using Unity.Collections;
+
+namespace Ecosystem.ECS.Events
+{
+    /// <summary>
+    /// Kills desired entities
+    /// </summary>
+    public class DeathEventSystem : ComponentSystem
+    {
+        protected override void OnUpdate()
+        {
+
+            Entities.WithAll<DeathCommand>().ForEach((Entity entity, /*int entityInQueryIndex,*/
+                ref DeathCommand deathCmd) =>
```

Comment:
When you only read from a component and don't modify it, you should use "in" instead of "ref"
```suggestion
                in DeathCommand deathCmd) =>
```